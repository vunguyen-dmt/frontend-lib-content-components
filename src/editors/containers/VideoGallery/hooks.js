import React from 'react';
import { useSelector } from 'react-redux';
import * as module from './hooks';
import messages from './messages';
import * as appHooks from '../../hooks';
import { selectors } from '../../data/redux';
import analyticsEvt from '../../data/constants/analyticsEvt';
import {
  filterKeys,
  filterMessages,
  sortKeys,
  sortMessages,
  sortFunctions,
} from './utils';

export const {
  navigateCallback,
  navigateTo,
} = appHooks;

export const state = {
  highlighted: (val) => React.useState(val),
  searchString: (val) => React.useState(val),
  showSelectVideoError: (val) => React.useState(val),
  showSizeError: (val) => React.useState(val),
  sortBy: (val) => React.useState(val),
  filertBy: (val) => React.useState(val),
  hideSelectedVideos: (val) => React.useState(val),
};

export const searchAndSortProps = () => {
  const [searchString, setSearchString] = module.state.searchString('');
  const [sortBy, setSortBy] = module.state.sortBy(sortKeys.dateNewest);
  const [filterBy, setFilterBy] = module.state.filertBy(filterKeys.videoStatus);
  const [hideSelectedVideos, setHideSelectedVideos] = module.state.hideSelectedVideos(false);

  return {
    searchString,
    onSearchChange: (e) => setSearchString(e.target.value),
    clearSearchString: () => setSearchString(''),
    sortBy,
    onSortClick: (key) => () => setSortBy(key),
    sortKeys,
    sortMessages,
    filterBy,
    onFilterClick: (key) => () => setFilterBy(key),
    filterKeys,
    filterMessages,
    showSwitch: true,
    hideSelectedVideos,
    switchMessage: messages.hideSelectedCourseVideosSwitchLabel,
    onSwitchClick: () => setHideSelectedVideos(!hideSelectedVideos),
  };
};

export const filterListBySearch = ({ searchString, videoList }) => (
  videoList.filter(({ displayName }) => displayName.toLowerCase().includes(searchString.toLowerCase()))
);

export const filterListByStatus = ({ statusFilter, videoList }) => {
  if (statusFilter === filterKeys.videoStatus) {
    return videoList;
  }
  return videoList.filter(({ status }) => status === statusFilter);
};

export const filterListByHideSelectedCourse = ({ videoList }) => (
  // TODO Missing to implement this
  videoList
);

export const filterList = ({
  sortBy,
  filterBy,
  searchString,
  videos,
}) => {
  let filteredList = module.filterListBySearch({
    searchString,
    videoList: videos,
  });
  filteredList = module.filterListByStatus({
    statusFilter: filterBy,
    videoList: filteredList,
  });
  filteredList = module.filterListByHideSelectedCourse({
    videoList: filteredList,
  });
  return filteredList.sort(sortFunctions[sortBy in sortKeys ? sortKeys[sortBy] : sortKeys.dateNewest]);
};

export const videoListProps = ({ searchSortProps, videos }) => {
  const [highlighted, setHighlighted] = module.state.highlighted(null);
  const [
    showSelectVideoError,
    setShowSelectVideoError,
  ] = module.state.showSelectVideoError(false);
  const [
    showSizeError,
    setShowSizeError,
  ] = module.state.showSizeError(false);
  const filteredList = module.filterList({ ...searchSortProps, videos });
  const learningContextId = useSelector(selectors.app.learningContextId);
  const blockId = useSelector(selectors.app.blockId);
  return {
    galleryError: {
      show: showSelectVideoError,
      set: () => setShowSelectVideoError(true),
      dismiss: () => setShowSelectVideoError(false),
      message: messages.selectVideoError,
    },
    // TODO We need to update this message when implementing the video upload screen
    inputError: {
      show: showSizeError,
      set: () => setShowSizeError(true),
      dismiss: () => setShowSelectVideoError(false),
      message: messages.fileSizeError,
    },
    galleryProps: {
      galleryIsEmpty: Object.keys(filteredList).length === 0,
      searchIsEmpty: filteredList.length === 0,
      displayList: filteredList,
      highlighted,
      onHighlightChange: (e) => setHighlighted(e.target.value),
      emptyGalleryLabel: messages.emptyGalleryLabel,
      showIdsOnCards: true,
      height: '100%',
    },
    selectBtnProps: {
      onClick: () => {
        if (highlighted) {
          navigateTo(`/course/${learningContextId}/editor/video/${blockId}?selectedVideoId=${highlighted}`);
        } else {
          setShowSelectVideoError(true);
        }
      },
    },
  };
};

export const fileInputProps = () => {
  const click = module.handleVideoUpload();
  return {
    click,
  };
};

export const handleVideoUpload = () => {
  const learningContextId = useSelector(selectors.app.learningContextId);
  const blockId = useSelector(selectors.app.blockId);
  return () => navigateTo(`/course/${learningContextId}/editor/video_upload/${blockId}`);
};

export const handleCancel = () => (
  navigateCallback({
    destination: useSelector(selectors.app.returnUrl),
    analytics: useSelector(selectors.app.analytics),
    analyticsEvent: analyticsEvt.videoGalleryCancelClick,
  })
);

export const buildVideos = ({ rawVideos }) => {
  let videos = [];
  const rawVideoList = Object.values(rawVideos);
  if (rawVideoList.length > 0) {
    videos = rawVideoList.map(video => ({
      id: video.edx_video_id,
      displayName: video.client_video_id,
      externalUrl: video.course_video_image_url,
      dateAdded: video.created,
      locked: false,
      thumbnail: video.course_video_image_url,
      status: video.status,
      statusBadgeVariant: module.getstatusBadgeVariant({ status: video.status }),
      duration: video.duration,
      transcripts: video.transcripts,
    }));
  }
  return videos;
};

export const getstatusBadgeVariant = ({ status }) => {
  switch (status) {
    case filterKeys.failed:
      return 'danger';
    case filterKeys.uploading:
    case filterKeys.processing:
      return 'light';
    default:
      return null;
  }
};

export const videoProps = ({ videos }) => {
  const searchSortProps = module.searchAndSortProps();
  const videoList = module.videoListProps({ searchSortProps, videos });
  const {
    galleryError,
    galleryProps,
    inputError,
    selectBtnProps,
  } = videoList;
  const fileInput = module.fileInputProps();

  return {
    galleryError,
    inputError,
    fileInput,
    galleryProps,
    searchSortProps,
    selectBtnProps,
  };
};

export default {
  videoProps,
  buildVideos,
  handleCancel,
  handleVideoUpload,
};
