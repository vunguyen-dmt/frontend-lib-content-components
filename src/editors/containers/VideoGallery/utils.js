import { StrictDict, keyStore } from '../../utils';
import messages from './messages';

const messageKeys = keyStore(messages);

export const sortKeys = StrictDict({
  dateNewest: 'dateNewest',
  dateOldest: 'dateOldest',
  nameAscending: 'nameAscending',
  nameDescending: 'nameDescending',
  durationShortest: 'durationShortest',
  durationLongest: 'durationLongest',
});

export const sortMessages = StrictDict({
  dateNewest: messages[messageKeys.sortByDateNewest],
  dateOldest: messages[messageKeys.sortByDateOldest],
  nameAscending: messages[messageKeys.sortByNameAscending],
  nameDescending: messages[messageKeys.sortByNameDescending],
  durationShortest: messages[messageKeys.sortByDurationShortest],
  durationLongest: messages[messageKeys.sortByDurationLongest],
});

export const filterKeys = StrictDict({
  videoStatus: 'videoStatus',
  uploading: 'uploading',
  processing: 'processing',
  ready: 'ready',
  failed: 'failed',
});

export const filterMessages = StrictDict({
  videoStatus: messages[messageKeys.filterByVideoStatusNone],
  uploading: messages[messageKeys.filterByVideoStatusUploading],
  processing: messages[messageKeys.filterByVideoStatusProcessing],
  ready: messages[messageKeys.filterByVideoStatusReady],
  failed: messages[messageKeys.filterByVideoStatusFailed],
});

export const sortFunctions = StrictDict({
  dateNewest: (a, b) => b.dateAdded - a.dateAdded,
  dateOldest: (a, b) => a.dateAdded - b.dateAdded,
  nameAscending: (a, b) => {
    const nameA = a.displayName.toLowerCase();
    const nameB = b.displayName.toLowerCase();
    if (nameA < nameB) { return -1; }
    if (nameB < nameA) { return 1; }
    return b.dateAdded - a.dateAdded;
  },
  nameDescending: (a, b) => {
    const nameA = a.displayName.toLowerCase();
    const nameB = b.displayName.toLowerCase();
    if (nameA < nameB) { return 1; }
    if (nameB < nameA) { return -1; }
    return b.dateAdded - a.dateAdded;
  },
  durationShortest: (a, b) => a.duration - b.duration,
  durationLongest: (a, b) => b.duration - a.duration,
});

export const acceptedImgKeys = StrictDict({
  mp4: '.mp4',
});
