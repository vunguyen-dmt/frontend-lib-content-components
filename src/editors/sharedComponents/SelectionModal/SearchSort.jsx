import React from 'react';
import PropTypes from 'prop-types';

import {
  ActionRow, Dropdown, Form, Icon, IconButton,
} from '@edx/paragon';
import { Close, Search } from '@edx/paragon/icons';
import {
  FormattedMessage,
  injectIntl,
  intlShape,
} from '@edx/frontend-platform/i18n';

import messages from './messages';

export const SearchSort = ({
  searchString,
  onSearchChange,
  clearSearchString,
  sortBy,
  onSortClick,
  sortKeys,
  sortMessages,
  filterBy,
  onFilterClick,
  filterKeys,
  filterMessages,
  showSwitch,
  switchMessage,
  onSwitchClick,
  // injected
  intl,
}) => (
  <ActionRow>
    <Form.Group style={{ margin: 0 }}>
      <Form.Control
        autoFocus
        onChange={onSearchChange}
        placeholder={intl.formatMessage(messages.searchPlaceholder)}
        trailingElement={
            searchString
              ? (
                <IconButton
                  iconAs={Icon}
                  invertColors
                  isActive
                  onClick={clearSearchString}
                  size="sm"
                  src={Close}
                />
              )
              : <Icon src={Search} />
          }
        value={searchString}
      />
    </Form.Group>

    { !showSwitch && <ActionRow.Spacer /> }
    <Dropdown>
      <Dropdown.Toggle className="text-gray-700" id="gallery-sort-button" variant="tertiary">
        <FormattedMessage {...sortMessages[sortBy]} />
      </Dropdown.Toggle>
      <Dropdown.Menu>
        {Object.keys(sortKeys).map(key => (
          <Dropdown.Item key={key} onClick={onSortClick(key)}>
            <FormattedMessage {...sortMessages[key]} />
          </Dropdown.Item>
        ))}
      </Dropdown.Menu>
    </Dropdown>

    { filterKeys && filterMessages && (
      <Dropdown>
        <Dropdown.Toggle className="text-gray-700" id="gallery-filter-button" variant="tertiary">
          <FormattedMessage {...filterMessages[filterBy]} />
        </Dropdown.Toggle>
        <Dropdown.Menu>
          {Object.keys(filterKeys).map(key => (
            <Dropdown.Item key={key} onClick={onFilterClick(key)}>
              <FormattedMessage {...filterMessages[key]} />
            </Dropdown.Item>
          ))}
        </Dropdown.Menu>
      </Dropdown>
    )}

    { showSwitch && (
      <>
        <ActionRow.Spacer />
        <Form.SwitchSet
          name="switch"
          onChange={onSwitchClick}
          isInline
        >
          <Form.Switch className="text-gray-700" value="switch-value" floatLabelLeft>
            <FormattedMessage {...switchMessage} />
          </Form.Switch>
        </Form.SwitchSet>
      </>
    )}

  </ActionRow>
);

SearchSort.defaultProps = {
  filterBy: '',
  onFilterClick: null,
  filterKeys: null,
  filterMessages: null,
  showSwitch: false,
  onSwitchClick: null,
};

SearchSort.propTypes = {
  searchString: PropTypes.string.isRequired,
  onSearchChange: PropTypes.func.isRequired,
  clearSearchString: PropTypes.func.isRequired,
  sortBy: PropTypes.string.isRequired,
  onSortClick: PropTypes.func.isRequired,
  sortKeys: PropTypes.shape({}).isRequired,
  sortMessages: PropTypes.shape({}).isRequired,
  filterBy: PropTypes.string,
  onFilterClick: PropTypes.func,
  filterKeys: PropTypes.shape({}),
  filterMessages: PropTypes.shape({}),
  showSwitch: PropTypes.bool,
  switchMessage: PropTypes.shape({}).isRequired,
  onSwitchClick: PropTypes.func,
  // injected
  intl: intlShape.isRequired,
};

export default injectIntl(SearchSort);
