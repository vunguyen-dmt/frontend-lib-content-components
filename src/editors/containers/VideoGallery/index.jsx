import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { selectors } from '../../data/redux';
import hooks from './hooks';
import SelectionModal from '../../sharedComponents/SelectionModal';
import { acceptedImgKeys } from './utils';
import messages from './messages';
import { RequestKeys } from '../../data/constants/requests';

export const VideoGallery = ({
  // redux
  rawVideos,
  isLoaded,
  isFetchError,
  isUploadError,
}) => {
  const videos = hooks.buildVideos({ rawVideos });
  const handleVideoUpload = hooks.handleVideoUpload();

  useEffect(() => {
    // If no videos exists redirects to the video upload screen
    if (isLoaded && videos.length === 0) {
      handleVideoUpload();
    }
  }, [isLoaded]);
  const {
    galleryError,
    inputError,
    fileInput,
    galleryProps,
    searchSortProps,
    selectBtnProps,
  } = hooks.videoProps({ videos });
  const handleCancel = hooks.handleCancel();

  const modalMessages = {
    confirmMsg: messages.selectVideoButtonlabel,
    titleMsg: messages.titleLabel,
    uploadButtonMsg: messages.uploadButtonLabel,
    fetchError: messages.fetchVideosError,
    uploadError: messages.uploadVideoError,
  };

  return (
    <div>
      <SelectionModal
        {...{
          isOpen: true,
          close: handleCancel,
          size: 'fullscreen',
          isFullscreenScroll: false,
          galleryError,
          inputError,
          fileInput,
          galleryProps,
          searchSortProps,
          selectBtnProps,
          acceptedFiles: acceptedImgKeys,
          modalMessages,
          isLoaded,
          isUploadError,
          isFetchError,
        }}
      />
    </div>
  );
};

VideoGallery.propTypes = {
  rawVideos: PropTypes.shape({}).isRequired,
  isLoaded: PropTypes.bool.isRequired,
  isFetchError: PropTypes.bool.isRequired,
  isUploadError: PropTypes.bool.isRequired,
};

export const mapStateToProps = (state) => ({
  rawVideos: selectors.app.videos(state),
  isLoaded: selectors.requests.isFinished(state, { requestKey: RequestKeys.fetchVideos }),
  isFetchError: selectors.requests.isFailed(state, { requestKey: RequestKeys.fetchVideos }),
  isUploadError: selectors.requests.isFailed(state, { requestKey: RequestKeys.uploadVideo }),
});

export const mapDispatchToProps = {};

export default connect(mapStateToProps, mapDispatchToProps)(VideoGallery);
