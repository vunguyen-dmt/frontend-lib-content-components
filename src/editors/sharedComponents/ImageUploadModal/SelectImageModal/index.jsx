import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import hooks from './hooks';
import { acceptedImgKeys } from './utils';
import SelectionModal from '../../SelectionModal';
import messages from './messages';
import { RequestKeys } from '../../../data/constants/requests';
import { selectors } from '../../../data/redux';

export const SelectImageModal = ({
  isOpen,
  close,
  setSelection,
  clearSelection,
  images,
  // redux
  isLoaded,
  isFetchError,
  isUploadError,
}) => {
  const {
    galleryError,
    inputError,
    fileInput,
    galleryProps,
    searchSortProps,
    selectBtnProps,
  } = hooks.imgHooks({ setSelection, clearSelection, images: images.current });

  const modalMessages = {
    confirmMsg: messages.nextButtonLabel,
    titleMsg: messages.titleLabel,
    uploadButtonMsg: messages.uploadButtonLabel,
    fetchError: messages.fetchImagesError,
    uploadError: messages.uploadImageError,
  };

  return (
    <SelectionModal
      {...{
        isOpen,
        close,
        galleryError,
        inputError,
        fileInput,
        galleryProps,
        searchSortProps,
        selectBtnProps,
        acceptedFiles: acceptedImgKeys,
        modalMessages,
        isLoaded,
        isFetchError,
        isUploadError,
      }}
    />
  );
};

SelectImageModal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  close: PropTypes.func.isRequired,
  setSelection: PropTypes.func.isRequired,
  clearSelection: PropTypes.func.isRequired,
  images: PropTypes.arrayOf(PropTypes.string).isRequired,
  // redux
  isLoaded: PropTypes.bool.isRequired,
  isFetchError: PropTypes.bool.isRequired,
  isUploadError: PropTypes.bool.isRequired,
};

export const mapStateToProps = (state) => ({
  isLoaded: selectors.requests.isFinished(state, { requestKey: RequestKeys.fetchAssets }),
  isFetchError: selectors.requests.isFailed(state, { requestKey: RequestKeys.fetchAssets }),
  isUploadError: selectors.requests.isFailed(state, { requestKey: RequestKeys.uploadAsset }),
});

export const mapDispatchToProps = {};

export default connect(mapStateToProps, mapDispatchToProps)(SelectImageModal);
