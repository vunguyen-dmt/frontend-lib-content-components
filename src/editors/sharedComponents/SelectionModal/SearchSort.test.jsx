import React from 'react';
import { shallow } from 'enzyme';
import { Dropdown } from '@edx/paragon';
import { FormattedMessage } from '@edx/frontend-platform/i18n';

import { formatMessage } from '../../../testUtils';

import { sortKeys, sortMessages } from '../ImageUploadModal/SelectImageModal/utils';
import { filterKeys, filterMessages } from '../../containers/VideoGallery/utils';
import { SearchSort } from './SearchSort';

describe('SearchSort component', () => {
  describe('snapshots without filterKeys', () => {
    const props = {
      searchString: 'props.searchString',
      onSearchChange: jest.fn().mockName('props.onSearchChange'),
      clearSearchString: jest.fn().mockName('props.clearSearchString'),
      sortBy: sortKeys.dateOldest,
      sortKeys,
      sortMessages,
      onSortClick: jest.fn().mockName('props.onSortClick'),
      intl: { formatMessage },
    };
    test('with search string (close button)', () => {
      expect(shallow(<SearchSort {...props} />)).toMatchSnapshot();
    });
    test('without search string (search icon)', () => {
      expect(shallow(<SearchSort {...props} searchString="" />)).toMatchSnapshot();
    });
    test('adds a sort option for each sortKey', () => {
      const el = shallow(<SearchSort {...props} />);
      expect(el.find(Dropdown).containsMatchingElement(
        <FormattedMessage {...sortMessages.dateNewest} />,
      )).toEqual(true);
      expect(el.find(Dropdown).containsMatchingElement(
        <FormattedMessage {...sortMessages.dateOldest} />,
      )).toEqual(true);
      expect(el.find(Dropdown).containsMatchingElement(
        <FormattedMessage {...sortMessages.nameAscending} />,
      )).toEqual(true);
      expect(el.find(Dropdown).containsMatchingElement(
        <FormattedMessage {...sortMessages.nameDescending} />,
      )).toEqual(true);
    });
  });
  describe('snapshots with filterKeys', () => {
    const props = {
      searchString: 'props.searchString',
      onSearchChange: jest.fn().mockName('props.onSearchChange'),
      clearSearchString: jest.fn().mockName('props.clearSearchString'),
      sortBy: sortKeys.dateOldest,
      sortKeys,
      sortMessages,
      filterKeys,
      filterMessages,
      showSwitch: true,
      onSortClick: jest.fn().mockName('props.onSortClick'),
      onFilterClick: jest.fn().mockName('props.onFilterClick'),
      intl: { formatMessage },
    };
    test('with search string (close button)', () => {
      expect(shallow(<SearchSort {...props} />)).toMatchSnapshot();
    });
    test('without search string (search icon)', () => {
      expect(shallow(<SearchSort {...props} searchString="" />)).toMatchSnapshot();
    });
    test('adds a sort option for each sortKey', () => {
      const el = shallow(<SearchSort {...props} />);
      expect(el.find(Dropdown).containsMatchingElement(
        <FormattedMessage {...sortMessages.dateNewest} />,
      )).toEqual(true);
      expect(el.find(Dropdown).containsMatchingElement(
        <FormattedMessage {...sortMessages.dateOldest} />,
      )).toEqual(true);
      expect(el.find(Dropdown).containsMatchingElement(
        <FormattedMessage {...sortMessages.nameAscending} />,
      )).toEqual(true);
      expect(el.find(Dropdown).containsMatchingElement(
        <FormattedMessage {...sortMessages.nameDescending} />,
      )).toEqual(true);
    });
    test('adds a filter option for each filterKet', () => {
      const el = shallow(<SearchSort {...props} />);
      expect(el.find(Dropdown).containsMatchingElement(
        <FormattedMessage {...filterMessages.videoStatus} />,
      )).toEqual(true);
      expect(el.find(Dropdown).containsMatchingElement(
        <FormattedMessage {...filterMessages.uploading} />,
      )).toEqual(true);
      expect(el.find(Dropdown).containsMatchingElement(
        <FormattedMessage {...filterMessages.processing} />,
      )).toEqual(true);
      expect(el.find(Dropdown).containsMatchingElement(
        <FormattedMessage {...filterMessages.ready} />,
      )).toEqual(true);
      expect(el.find(Dropdown).containsMatchingElement(
        <FormattedMessage {...filterMessages.failed} />,
      )).toEqual(true);
    });
  });
});
