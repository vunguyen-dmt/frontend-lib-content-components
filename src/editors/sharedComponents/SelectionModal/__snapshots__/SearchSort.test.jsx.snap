// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSort component snapshots with filterKeys with search string (close button) 1`] = `
<ActionRow>
  <Form.Group
    style={
      Object {
        "margin": 0,
      }
    }
  >
    <Form.Control
      autoFocus={true}
      onChange={[MockFunction props.onSearchChange]}
      placeholder="Search"
      trailingElement={
        <IconButton
          iconAs="Icon"
          invertColors={true}
          isActive={true}
          onClick={[MockFunction props.clearSearchString]}
          size="sm"
          src={[MockFunction icons.Close]}
        />
      }
      value="props.searchString"
    />
  </Form.Group>
  <Dropdown>
    <Dropdown.Toggle
      className="text-gray-700"
      id="gallery-sort-button"
      variant="tertiary"
    >
      <FormattedMessage
        defaultMessage="By date added (oldest)"
        description="Dropdown label for sorting by date (oldest)"
        id="authoring.texteditor.selectimagemodal.sort.dateoldest.label"
      />
    </Dropdown.Toggle>
    <Dropdown.Menu>
      <Dropdown.Item
        key="dateNewest"
      >
        <FormattedMessage
          defaultMessage="By date added (newest)"
          description="Dropdown label for sorting by date (newest)"
          id="authoring.texteditor.selectimagemodal.sort.datenewest.label"
        />
      </Dropdown.Item>
      <Dropdown.Item
        key="dateOldest"
      >
        <FormattedMessage
          defaultMessage="By date added (oldest)"
          description="Dropdown label for sorting by date (oldest)"
          id="authoring.texteditor.selectimagemodal.sort.dateoldest.label"
        />
      </Dropdown.Item>
      <Dropdown.Item
        key="nameAscending"
      >
        <FormattedMessage
          defaultMessage="By name (ascending)"
          description="Dropdown label for sorting by name (ascending)"
          id="authoring.texteditor.selectimagemodal.sort.nameascending.label"
        />
      </Dropdown.Item>
      <Dropdown.Item
        key="nameDescending"
      >
        <FormattedMessage
          defaultMessage="By name (descending)"
          description="Dropdown label for sorting by name (descending)"
          id="authoring.texteditor.selectimagemodal.sort.namedescending.label"
        />
      </Dropdown.Item>
    </Dropdown.Menu>
  </Dropdown>
  <Dropdown>
    <Dropdown.Toggle
      className="text-gray-700"
      id="gallery-filter-button"
      variant="tertiary"
    >
      <FormattedMessage />
    </Dropdown.Toggle>
    <Dropdown.Menu>
      <Dropdown.Item
        key="videoStatus"
      >
        <FormattedMessage
          defaultMessage="Video status"
          description="Dropdown label for filter by video status (none)"
          id="authoring.selectvideomodal.filter.videostatusnone.label"
        />
      </Dropdown.Item>
      <Dropdown.Item
        key="uploading"
      >
        <FormattedMessage
          defaultMessage="Uploading"
          description="Dropdown label for filter by video status (uploading)"
          id="authoring.selectvideomodal.filter.videostatusuploading.label"
        />
      </Dropdown.Item>
      <Dropdown.Item
        key="processing"
      >
        <FormattedMessage
          defaultMessage="Processing"
          description="Dropdown label for filter by video status (processing)"
          id="authoring.selectvideomodal.filter.videostatusprocessing.label"
        />
      </Dropdown.Item>
      <Dropdown.Item
        key="ready"
      >
        <FormattedMessage
          defaultMessage="Ready"
          description="Dropdown label for filter by video status (ready)"
          id="authoring.selectvideomodal.filter.videostatusready.label"
        />
      </Dropdown.Item>
      <Dropdown.Item
        key="failed"
      >
        <FormattedMessage
          defaultMessage="Failed"
          description="Dropdown label for filter by video status (failed)"
          id="authoring.selectvideomodal.filter.videostatusfailed.label"
        />
      </Dropdown.Item>
    </Dropdown.Menu>
  </Dropdown>
  <ActionRow.Spacer />
  <Component
    isInline={true}
    name="switch"
    onChange={null}
  >
    <Component
      className="text-gray-700"
      floatLabelLeft={true}
      value="switch-value"
    >
      <FormattedMessage />
    </Component>
  </Component>
</ActionRow>
`;

exports[`SearchSort component snapshots with filterKeys without search string (search icon) 1`] = `
<ActionRow>
  <Form.Group
    style={
      Object {
        "margin": 0,
      }
    }
  >
    <Form.Control
      autoFocus={true}
      onChange={[MockFunction props.onSearchChange]}
      placeholder="Search"
      trailingElement={<Icon />}
      value=""
    />
  </Form.Group>
  <Dropdown>
    <Dropdown.Toggle
      className="text-gray-700"
      id="gallery-sort-button"
      variant="tertiary"
    >
      <FormattedMessage
        defaultMessage="By date added (oldest)"
        description="Dropdown label for sorting by date (oldest)"
        id="authoring.texteditor.selectimagemodal.sort.dateoldest.label"
      />
    </Dropdown.Toggle>
    <Dropdown.Menu>
      <Dropdown.Item
        key="dateNewest"
      >
        <FormattedMessage
          defaultMessage="By date added (newest)"
          description="Dropdown label for sorting by date (newest)"
          id="authoring.texteditor.selectimagemodal.sort.datenewest.label"
        />
      </Dropdown.Item>
      <Dropdown.Item
        key="dateOldest"
      >
        <FormattedMessage
          defaultMessage="By date added (oldest)"
          description="Dropdown label for sorting by date (oldest)"
          id="authoring.texteditor.selectimagemodal.sort.dateoldest.label"
        />
      </Dropdown.Item>
      <Dropdown.Item
        key="nameAscending"
      >
        <FormattedMessage
          defaultMessage="By name (ascending)"
          description="Dropdown label for sorting by name (ascending)"
          id="authoring.texteditor.selectimagemodal.sort.nameascending.label"
        />
      </Dropdown.Item>
      <Dropdown.Item
        key="nameDescending"
      >
        <FormattedMessage
          defaultMessage="By name (descending)"
          description="Dropdown label for sorting by name (descending)"
          id="authoring.texteditor.selectimagemodal.sort.namedescending.label"
        />
      </Dropdown.Item>
    </Dropdown.Menu>
  </Dropdown>
  <Dropdown>
    <Dropdown.Toggle
      className="text-gray-700"
      id="gallery-filter-button"
      variant="tertiary"
    >
      <FormattedMessage />
    </Dropdown.Toggle>
    <Dropdown.Menu>
      <Dropdown.Item
        key="videoStatus"
      >
        <FormattedMessage
          defaultMessage="Video status"
          description="Dropdown label for filter by video status (none)"
          id="authoring.selectvideomodal.filter.videostatusnone.label"
        />
      </Dropdown.Item>
      <Dropdown.Item
        key="uploading"
      >
        <FormattedMessage
          defaultMessage="Uploading"
          description="Dropdown label for filter by video status (uploading)"
          id="authoring.selectvideomodal.filter.videostatusuploading.label"
        />
      </Dropdown.Item>
      <Dropdown.Item
        key="processing"
      >
        <FormattedMessage
          defaultMessage="Processing"
          description="Dropdown label for filter by video status (processing)"
          id="authoring.selectvideomodal.filter.videostatusprocessing.label"
        />
      </Dropdown.Item>
      <Dropdown.Item
        key="ready"
      >
        <FormattedMessage
          defaultMessage="Ready"
          description="Dropdown label for filter by video status (ready)"
          id="authoring.selectvideomodal.filter.videostatusready.label"
        />
      </Dropdown.Item>
      <Dropdown.Item
        key="failed"
      >
        <FormattedMessage
          defaultMessage="Failed"
          description="Dropdown label for filter by video status (failed)"
          id="authoring.selectvideomodal.filter.videostatusfailed.label"
        />
      </Dropdown.Item>
    </Dropdown.Menu>
  </Dropdown>
  <ActionRow.Spacer />
  <Component
    isInline={true}
    name="switch"
    onChange={null}
  >
    <Component
      className="text-gray-700"
      floatLabelLeft={true}
      value="switch-value"
    >
      <FormattedMessage />
    </Component>
  </Component>
</ActionRow>
`;

exports[`SearchSort component snapshots without filterKeys with search string (close button) 1`] = `
<ActionRow>
  <Form.Group
    style={
      Object {
        "margin": 0,
      }
    }
  >
    <Form.Control
      autoFocus={true}
      onChange={[MockFunction props.onSearchChange]}
      placeholder="Search"
      trailingElement={
        <IconButton
          iconAs="Icon"
          invertColors={true}
          isActive={true}
          onClick={[MockFunction props.clearSearchString]}
          size="sm"
          src={[MockFunction icons.Close]}
        />
      }
      value="props.searchString"
    />
  </Form.Group>
  <ActionRow.Spacer />
  <Dropdown>
    <Dropdown.Toggle
      className="text-gray-700"
      id="gallery-sort-button"
      variant="tertiary"
    >
      <FormattedMessage
        defaultMessage="By date added (oldest)"
        description="Dropdown label for sorting by date (oldest)"
        id="authoring.texteditor.selectimagemodal.sort.dateoldest.label"
      />
    </Dropdown.Toggle>
    <Dropdown.Menu>
      <Dropdown.Item
        key="dateNewest"
      >
        <FormattedMessage
          defaultMessage="By date added (newest)"
          description="Dropdown label for sorting by date (newest)"
          id="authoring.texteditor.selectimagemodal.sort.datenewest.label"
        />
      </Dropdown.Item>
      <Dropdown.Item
        key="dateOldest"
      >
        <FormattedMessage
          defaultMessage="By date added (oldest)"
          description="Dropdown label for sorting by date (oldest)"
          id="authoring.texteditor.selectimagemodal.sort.dateoldest.label"
        />
      </Dropdown.Item>
      <Dropdown.Item
        key="nameAscending"
      >
        <FormattedMessage
          defaultMessage="By name (ascending)"
          description="Dropdown label for sorting by name (ascending)"
          id="authoring.texteditor.selectimagemodal.sort.nameascending.label"
        />
      </Dropdown.Item>
      <Dropdown.Item
        key="nameDescending"
      >
        <FormattedMessage
          defaultMessage="By name (descending)"
          description="Dropdown label for sorting by name (descending)"
          id="authoring.texteditor.selectimagemodal.sort.namedescending.label"
        />
      </Dropdown.Item>
    </Dropdown.Menu>
  </Dropdown>
</ActionRow>
`;

exports[`SearchSort component snapshots without filterKeys without search string (search icon) 1`] = `
<ActionRow>
  <Form.Group
    style={
      Object {
        "margin": 0,
      }
    }
  >
    <Form.Control
      autoFocus={true}
      onChange={[MockFunction props.onSearchChange]}
      placeholder="Search"
      trailingElement={<Icon />}
      value=""
    />
  </Form.Group>
  <ActionRow.Spacer />
  <Dropdown>
    <Dropdown.Toggle
      className="text-gray-700"
      id="gallery-sort-button"
      variant="tertiary"
    >
      <FormattedMessage
        defaultMessage="By date added (oldest)"
        description="Dropdown label for sorting by date (oldest)"
        id="authoring.texteditor.selectimagemodal.sort.dateoldest.label"
      />
    </Dropdown.Toggle>
    <Dropdown.Menu>
      <Dropdown.Item
        key="dateNewest"
      >
        <FormattedMessage
          defaultMessage="By date added (newest)"
          description="Dropdown label for sorting by date (newest)"
          id="authoring.texteditor.selectimagemodal.sort.datenewest.label"
        />
      </Dropdown.Item>
      <Dropdown.Item
        key="dateOldest"
      >
        <FormattedMessage
          defaultMessage="By date added (oldest)"
          description="Dropdown label for sorting by date (oldest)"
          id="authoring.texteditor.selectimagemodal.sort.dateoldest.label"
        />
      </Dropdown.Item>
      <Dropdown.Item
        key="nameAscending"
      >
        <FormattedMessage
          defaultMessage="By name (ascending)"
          description="Dropdown label for sorting by name (ascending)"
          id="authoring.texteditor.selectimagemodal.sort.nameascending.label"
        />
      </Dropdown.Item>
      <Dropdown.Item
        key="nameDescending"
      >
        <FormattedMessage
          defaultMessage="By name (descending)"
          description="Dropdown label for sorting by name (descending)"
          id="authoring.texteditor.selectimagemodal.sort.namedescending.label"
        />
      </Dropdown.Item>
    </Dropdown.Menu>
  </Dropdown>
</ActionRow>
`;
